python/packages/autogen-core/docs/src/user-guide/core-user-guide/cookbook/distributed-group-chat.ipynb

```diff
@@ -0,0 +1,258 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Distributed Round Robin Group Chat\n",
+    "This notebook demonstrates how to implement a distributed round-robin group chat. In this example we will create three {py:class} `WorkerAgentRuntime` runtime object, two chat participant and a group chat manager, and each of them will connect to an instance of  `WorkerAgentRuntimeHost` which implements the required grpc server.\n",
+    "You would only need to provide `OPENAI_API_KEY` variable below to be able to run this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "OPENAI_API_KEY = os.environ[\"OPENAI_API_KEY\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import asyncio\n",
+    "from typing import List\n",
+    "\n",
+    "from autogen_core.application import WorkerAgentRuntime, WorkerAgentRuntimeHost\n",
+    "from autogen_core.base import MessageContext\n",
+    "from autogen_core.base._serialization import try_get_known_serializers_for_type\n",
+    "from autogen_core.base._topic import TopicId\n",
+    "from autogen_core.components import (\n",
+    "    RoutedAgent,\n",
+    "    message_handler,\n",
+    "    type_subscription,\n",
+    ")\n",
+    "from autogen_core.components._default_subscription import DefaultSubscription\n",
+    "from autogen_core.components._type_subscription import TypeSubscription\n",
+    "from autogen_core.components.models import (\n",
+    "    AssistantMessage,\n",
+    "    ChatCompletionClient,\n",
+    "    LLMMessage,\n",
+    "    OpenAIChatCompletionClient,\n",
+    "    SystemMessage,\n",
+    "    UserMessage,\n",
+    ")\n",
+    "from pydantic import BaseModel\n",
+    "\n",
+    "\n",
+    "class GroupChatMessage(BaseModel):\n",
+    "    \"\"\"Implements a sample message sent by an LLM agent\"\"\"\n",
+    "\n",
+    "    body: LLMMessage\n",
+    "\n",
+    "\n",
+    "class RequestToSpeak(BaseModel):\n",
+    "    \"\"\"Message type for agents to speak\"\"\"\n",
+    "\n",
+    "    pass"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Group Chat Manager\n",
+    "- **group chat manager** implements a round-robing algorithm to ask agents to talk, one at a time. It sends `SpeakRequest` requests to agent topics. This agent subscrives to topic type **agent_result** to receive participants messages.\n",
+    "\n",
+    "# Participants\n",
+    "- **data scientist**: Listens to `SpeakRequest` sent by manager and publishes a LLM API call response as `GroupChatMessage`. This agent subscribes to `topic_type=\"data_scientist\", agent_type=\"data_scientist_ag_type\"`\n",
```

Comment:
all agents should subscribe to a common topic type for the group chat. See: https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/design-patterns/group-chat.html